### ISSUE TITLE
Add ability to group logs

### ISSUE THREAD

**Issue by garygreen** [stevebauman/log-reader#3]
In combination with #2 it would be cool to be able to group logs by a closure or array of attributes. For instance:

```php
LogReader::groupBy('class');
LogReader::groupBy(['message', 'class']);
```

This will make it much nicer to output a list of logs with related exceptions / issues.

**Comment by stevebauman** [comment]
This is true, but you're always able to call the resulting collections `groupBy()` method from a `LogReader::get()`. Not sure if there's value in this since all logs would need to be gathered before hand anyway.

Though an alias might be handy which would call the `get()` method and then the resulting `groupBy()` collection method.
